### PR TITLE
Chore: Replace entity GRN with infra/grn GRN

### PR DESCRIPTION
**Pull request by DanCech**
This PR replaces the entity definition of GRN with the definition from the infra/grn package, and updates the infra/grn package to use a proto definition so that we can import it into the entity proto definition. This allows us to have a single unified understanding of what a GRN is.

To support this I did have to change infra/grn somewhat, the main changes being to use pointers to the GRN structs so that we aren't copying the internal mutexes used by protobuf, and changing the String function to ToGRNString because proto defines String already.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -263,6 +263,9 @@ devenv-mysql:
 protobuf: ## Compile protobuf definitions
 	bash scripts/protobuf-check.sh
 	bash pkg/plugins/backendplugin/pluginextensionv2/generate.sh
+	bash pkg/plugins/backendplugin/secretsmanagerplugin/generate.sh
+	bash pkg/services/store/entity/generate.sh
+	bash pkg/infra/grn/generate.sh
```
Comment (zserge) [inline Makefile:268]: Shall we maybe have a script that `find`'s all `.proto` files and applies `protoc` command to them?

File: pkg/infra/grn/grn.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	// an organization.
+	int64 TenantID = 1;
+
+	// The kind of resource being identified, for e.g. "dashboard" or "user".
+	// The caller is responsible for validating the value.
+	string ResourceKind = 3;
+
+	// ResourceIdentifier is used by the underlying service to identify the
+	// resource.
+	string ResourceIdentifier = 4;
```
Comment (zserge) [inline pkg/infra/grn/grn.proto:19]: If this is a relatively new and unexposed message - why do we use 1, 3 and 4 and skip the 2 for field IDs?

**Review by zserge — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (DanCech) [reply Makefile:268]: There is some nuance to the generate.sh files right now, but we could potentially do something like that if we can resolve the differences that we need currently. That sounds like a separate PR though ;)

**Review by mildwonkey — APPROVED** [review]
LGTM! (I also quite appreciate `ResourceKind` and `ResourceIdentifier`, I find those much clearer)

**Pull request merged**